substrate/client/offchain/src/api/http.rs

```diff
@@ -48,21 +48,26 @@ use std::{
 
 const LOG_TARGET: &str = "offchain-worker::http";
 
+pub type Body = BoxBody<hyper::body::Bytes, hyper::Error>;
+
+type Sender = mpsc::Sender<Result<hyper::body::Frame<hyper::body::Bytes>, hyper::Error>>;
+type Receiver = mpsc::Receiver<Result<hyper::body::Frame<hyper::body::Bytes>, hyper::Error>>;
+
+type HyperClient = client::Client<HttpsConnector<client::connect::HttpConnector>, Body>;
+type LazyClient = Lazy<HyperClient, Box<dyn FnOnce() -> HyperClient + Send>>;
+
 /// Wrapper struct used for keeping the hyper_rustls client running.
 #[derive(Clone)]
-pub struct SharedClient(Arc<Lazy<HyperClient<HttpsConnector<client::HttpConnector>, Body>>>);
+pub struct SharedClient(Arc<LazyClient>);
 
 impl SharedClient {
-	pub fn new() -> Self {
-		Self(Arc::new(Lazy::new(|| {
-			let connector = HttpsConnectorBuilder::new()
-				.with_native_roots()
-				.https_or_http()
-				.enable_http1()
-				.enable_http2()
-				.build();
-			HyperClient::builder().build(connector)
-		})))
+	pub fn new() -> std::io::Result<Self> {
+		let builder = HttpsConnectorBuilder::new()
+			.with_provider_and_native_roots(rustls::crypto::ring::default_provider())?;
```

Comment:
```suggestion
			.with_native_roots()?;
```

Reply:
It seems that recent `rustls` versions are panicking without provided or process-wide crypto provider
https://github.com/paritytech/polkadot-sdk/actions/runs/11173098681/job/31131298329

```
--- TRY 1 STDERR:        sc-offchain api::http::tests::request_add_header_invalid_call ---
thread 'tokio-runtime-worker' panicked at /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/rustls-0.23.10/src/crypto/mod.rs:259:14:
no process-level CryptoProvider available -- call CryptoProvider::install_default() before this point
```

Should I set it in test code with [CryptoProvider::install_default()](https://docs.rs/rustls/latest/rustls/crypto/struct.CryptoProvider.html#method.install_default) and leave calling it to user in the library level?